config-example.yml

```diff
@@ -20,6 +20,50 @@ graphhopper:
   # More are: surface,max_width,max_height,max_weight,max_axle_load,max_length,hazmat,hazmat_tunnel,hazmat_water,toll,track_type
   # graph.encoded_values: surface,toll,track_type
 
+  ##### Routing Profiles ####
+
+  # Routing can be done for the following list of profiles. The fields of each profile are as follows:
+  # - name (required): a unique string identifier for the profile
+  # - vehicle (required): refers to the `graph.flag_encoder` used for this profile
+  # - weighting (required): the weighting used for this profile, e.g. fastest,shortest or short_fastest
+  # - turn_costs (true/false, default: false): whether or not turn restrictions should be applied for this profile.
+  #   this will only work if the `graph.flag_encoder` for the given `vehicle` is configured with `|turn_costs=true`.
```

Comment:
Yes it might be possible, but lets say this would be another/next step (which is not strictly required here). The problem I am having with this is that the flag encoders are not simply something like `car`, but rather `car|turn_costs=true` or `motorcycle|your_property=here`. So right now if we would do any kind of such configurations of the flag encoders like this we would have to repeat it in each profile? Or somehow merge them? Or choose one by some (possibley not so obvious) rule? And even if we think a bit about future development when the flag encoders become collections of encoded values, then we will probably want to configure these collections of encoded values, give them a name and refer to this name in the profile (just like we are doing here).

> btw minor typo of missing 's' in both cases

fixed, thanks

Reply:
Ok, yeah. Probably for another issue

> The problem I am having with this is that the flag encoders are not simply something like car, but rather car|turn_costs=true or motorcycle|your_property=here

Could this be done via

```
profile:
 - name: car
 - vehicle: car
 - your_property: your_value
```
?

Reply:
If we assume there is one encoder per profile, thats the problem. Otherwise it would be unclear whether the (one) car encoder would have turn instructions or not (and probably similar problems for other encoder properties), there would no longer be a single place to define the encoder.